file and python dependents: add missing dependencies

sethrj opened the pull request:
- Mark redundant python-magic package
- Add dependencies on the `file` package: note that these libraries import `libmagic.{so,dylib,dll}` at runtime rather than calling the `file` command, so not all external installations of `file` will work here!
- Add compressor dependencies to `file`; previously it was linking against system compression libraries rather than spack ones.

sethrj committed:
py-magic: delete redundant package

This package is actually named py-python-magic (since the project itself
is "python-magic").

sethrj committed:
New package: libmagic

sethrj committed:
Py-python-magic: add required runtime dependency on libmagic and new version

sethrj committed:
Py-filemagic: add required runtime dependency

adamjstewart commented on var/spack/repos/builtin/packages/py-magic/package.py:
```diff
@@ -1,18 +0,0 @@
-from spack import *
-
-
-class PyMagic(PythonPackage):
```
I agree that this is a duplicate package, but we should first deprecate it and remove it in a future release. Just add `deprecated=True` to the version.

sethrj replied:
Fair enough!

adamjstewart commented on var/spack/repos/builtin/packages/libmagic/package.py:
```diff
@@ -0,0 +1,57 @@
+    """
+
+    homepage = "https://www.darwinsys.com/file/"
+    url      = "https://astron.com/pub/file/file-5.40.tar.gz"
```
Isn't this a duplicate of our `file` package?

sethrj replied:
Well darn, I should have checked that first! Yes, will fix.

sethrj committed:
py-magic: restore and mark as redundant

This reverts commit 4cab7fb69e2c2b8098895bee92aabe5df8b7aaaa.

sethrj committed:
file: add implicit dependencies and static variant

Replaces redundant libmagic that I added. Compression headers were previously
being picked up from the system.

sethrj committed:
Fix py-python-magic dependency

adamjstewart commented on var/spack/repos/builtin/packages/py-python-magic/package.py:
```diff
@@ -8,11 +8,16 @@
 
+    version('0.4.24', sha256='de800df9fb50f8ec5974761054a708af6e4246b03b4bdaee993f948947b0ebcf')
     version('0.4.15', sha256='f3765c0f582d2dfc72c15f3b5a82aecfae9498bd29ca840d72f37d7bd38bfcd5')
 
```
Can you also add:
```python
depends_on('python@2.7.0:2.7,3.5:', type=('build', 'run'))
```
or whatever the magic syntax is these days?

sethrj committed:
Update python version requirements

adamjstewart commented on var/spack/repos/builtin/packages/file/package.py:
```diff
@@ -15,15 +15,34 @@ class File(AutotoolsPackage):
+        args = [
+            "--disable-dependency-tracking",
+            "--enable-fsect-man5",
+            "--enable-zlib",
+            "--enable-bzlib",
+            "--enable-xzlib",
```
If these are optional, we might want to add variants and make them off by default. Trying to minimize build times. But this isn't my package so if you're the maintainer you get to decide 😄

haampie replied:
some packages have a multivalues variant `compression=x,y,z` now btw.

sethrj replied:
Build times are quite small, and as I mentioned in the description the default was to turn them on and link against system libraries. I'm happy to review a PR for adding compression variants now that I'm maintainer ;)

adamjstewart replied:
Yeah, we def don't want to link to system libs regardless.
